plugin/path_rotate_root_creds.go

```diff
@@ -33,22 +37,66 @@ func (b *backend) pathRotateCredentialsUpdate(ctx context.Context, req *logical.
 	if err != nil {
 		return nil, err
 	}
+	oldPassword := engineConf.ADConf.BindPassword
 
+	if !atomic.CompareAndSwapInt32(b.rotateRootLock, 0, 1) {
+		b.Logger().Warn("rotate-root operation is already in progress")
+		// Respond with a 204 since this is not an error state.
+		return nil, nil
```

Comment:
Most of the other tidy endpoints respond with a 200 and provide a warning that it's already in progress. See for example https://github.com/hashicorp/vault/blob/master/builtin/credential/approle/path_tidy_user_id.go#L31

Reply:
Kk, will update!